File: Algorithms/BOJ/Python/5397.py
# https://www.acmicpc.net/problem/5397
# 키로거
import sys
n = int(sys.stdin.readline())
for _ in range(n):
    L = sys.stdin.readline().strip()
    left = []
    right = []
    for s in L:
        if s == "-":
            if left:
                left.pop()
        elif s == "<":
            if left:
                right.append(left.pop())
        elif s == ">":
            if right:
                left.append(right.pop())
        else: # character
            left.append(s)
    print(''.join(left) + ''.join(right[::-1]))

File: Algorithms/BOJ/Swift/5397.swift
// https://www.acmicpc.net/problem/5397
// 키로거
let n = Int(readLine()!)!
for _ in 0..<n {
    let str = readLine()!
    var left: [String] = []
    var right: [String] = []
    for s in str {
        if s == "-" {
            if !left.isEmpty {
                left.removeLast()
            }
        } else if s == "<" {
            if !left.isEmpty {
                right.append(left.removeLast())
            }
        } else if s == ">" {
            if !right.isEmpty {
                left.append(right.removeLast())
            }
        } else { // Character
            left.append(String(s))
        }
    }
    print(left.joined() + right.reversed().joined())
}